core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -3167,6 +3169,404 @@ public void testRegisterExistingTable() {
     assertThat(catalog.dropTable(identifier)).isTrue();
   }
 
+  @Test
+  public void testBranchLifecycleManagement() {
+    C catalog = catalog();
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(NS);
+    }
+
+    Table table = catalog.buildTable(TABLE, SCHEMA).create();
+    table.newFastAppend().appendFile(FILE_A).commit();
+    Snapshot firstSnapshot = table.currentSnapshot();
+
+    String branchName1 = "feature-branch-1";
+    table.manageSnapshots().createBranch(branchName1).commit();
+
+    // Verify branch was created
+    Map<String, SnapshotRef> refs = table.refs();
+    assertThat(refs).containsKey(branchName1);
+    assertThat(refs.get(branchName1).isBranch()).isTrue();
+    assertThat(refs.get(branchName1).snapshotId()).isEqualTo(firstSnapshot.snapshotId());
+
+    table.newFastAppend().appendFile(FILE_B).commit();
+    Snapshot secondSnapshot = table.currentSnapshot();
+
+    String branchName2 = "feature-branch-2";
+    table.manageSnapshots().createBranch(branchName2, firstSnapshot.snapshotId()).commit();
+
+    // Verify branch points to the specific snapshot
+    refs = table.refs();
+    assertThat(refs).containsKey(branchName2);
+    assertThat(refs.get(branchName2).snapshotId()).isEqualTo(firstSnapshot.snapshotId());
+    assertThat(refs.get(branchName2).snapshotId()).isNotEqualTo(secondSnapshot.snapshotId());
+
+    String branchName3 = "retention-branch";
+    int minSnapshotsToKeep = 5;
+    long maxSnapshotAgeMs = 7 * 24 * 60 * 60 * 1000L; // 7 days
```

Comment:
nit: We could use `Duration.ofDays(7).toMillis()` instead. Same for other millis variables.